### DISCUSSION TITLE
HTMX and jinja templates in single app

### DISCUSSION THREAD

**Discussion by Murtagy** [litestar-org/litestar#3426] [Q&A] [answered]
I want some my endpoints to respond with full pages, while others to support partial content updates with htmx. 
As far as I understand `template_config` on application accepts only 1 template engine and I can not stack applications on each other.

**Comment by JacobCoffee** [comment, marked as answer]
You want regular, full Jinja routes and then partial HTMX routes?
You can see https://github.com/JacobCoffee/niapi/tree/main/app/domain/web/templates for a crude example that uses both
- https://github.com/JacobCoffee/niapi/blob/main/app/domain/calculator/controllers/calculator.py

Basically for your full templates you would just return a `Template` but for the partials you want to do via HTMX you would return an [HTMXTemplate](https://docs.litestar.dev/latest/usage/htmx.html#htmxtemplate-response-classes)

HTMX is not its own templating engine, but can be used with any we provide or any custom ones (as you can see in my partial HTMX HTML file, it also uses Jinja to fill in some blanks)

**Reply by Murtagy** [reply]
```
app = Litestar(
    route_handlers=[get_form],
    debug=True,
    request_class=HTMXRequest,
    template_config=TemplateConfig(
        directory=Path("litestar_htmx/templates"),
        engine=JinjaTemplateEngine,
    ),
)
```
I guess the line of setting the request class globally for an app confused me.

Thank you for clarifications, this is answered
